.github/workflows/ci.yaml

```diff
@@ -9,19 +9,104 @@ on:
     types: [ published ]
 
 jobs:
+
+  lint:
+    name: Linter
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Cache PyPI
+        uses: actions/cache@v2
+        with:
+          key: pip-lint-${{ hashFiles('requirements/*.txt') }}
+          path: ~/.cache/pip
+          restore-keys: |
+              pip-lint-
+      - name: Install dependencies
+        run: |
+          python -m pip install twine wheel
+          make setup
+      - name: Run linters
+        run: |
+          make lint
+      - name: Save the package version
+        id: version
+        run: |
+          echo "::set-output name=version::$(python setup.py --version)"
+      - name: Show version
+        run: |
+            echo ${{ steps.version.outputs.version }}
+
+  pretest:
+    name: Prepare for e2e tests
+    needs: [lint]
+    if: github.event_name != 'release'
```

Comment:
I've missed a little.
Do you want skip pretest and e2e for releases?

Reply:
yes this was discussed